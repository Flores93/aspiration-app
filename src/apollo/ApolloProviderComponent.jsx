import { InMemoryCache, ApolloClient, ApolloProvider } from "@apollo/client";

const options = {
  uri: "https://api.github.com/graphql",
  cache: new InMemoryCache(),
  headers: {
    authorization: `bearer ${process.env.REACT_APP_GIT_TOKEN}`,
  },
};

const client = new ApolloClient(options);

const ApolloProviderComponent = ({ children }) => (
  <ApolloProvider client={client}>{children}</ApolloProvider>
);

export default ApolloProviderComponent;
