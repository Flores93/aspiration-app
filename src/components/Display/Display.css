.topics-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.topics-container > * {
  margin: 1em 0;
}
