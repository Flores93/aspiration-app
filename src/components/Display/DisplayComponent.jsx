import { useQuery } from "@apollo/client";
import { GET_TOPICS } from "../../apollo/gqlQueries";

import Topic from "../Topic";
import "./Display.css";

const Display = ({ searchValue, setsearchValue }) => {
  const gqlVariables = {
    variables: { searchValue },
  };
  const { loading, error, data } = useQuery(GET_TOPICS, gqlVariables);

  if (loading)
    return (
      <p style={{ fontSize: "2rem" }}>
        Loading...
        <i className="fa fa-star fa-spin"></i>
      </p>
    );

  if (error) return `Something went wrong: ${error}`;

  return (
    <main>
      <div className="topics-container">
        {data && data.search.edges
          ? data.search.edges.map((edge, index) => (
              <Topic key={index} edge={edge} changeTopic={setsearchValue} />
            ))
          : null}
      </div>
    </main>
  );
};

export default Display;
