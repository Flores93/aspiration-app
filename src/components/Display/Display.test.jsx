import { mount } from "enzyme";
import ApolloProviderComponent from "../../apollo/ApolloProviderComponent";
import Display from "./DisplayComponent";

describe("Display Component Test", () => {
  const setSearchValueMock = jest.fn();

  test("should show the component correctly with the loading spinning star", () => {
    const wrapper = mount(
      <ApolloProviderComponent>
        <Display searchValue="React" setsearchValue={setSearchValueMock} />
      </ApolloProviderComponent>
    );

    const findStarSpinning = wrapper.find(".fa-spin").exists();

    expect(findStarSpinning).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
