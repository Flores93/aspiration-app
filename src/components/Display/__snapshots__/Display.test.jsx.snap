// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component Test should show the component correctly with the loading spinning star 1`] = `
<p
  style={
    Object {
      "fontSize": "2rem",
    }
  }
>
  Loading...
  <i
    className="fa fa-star fa-spin"
  />
</p>
`;
